src/main/java/org/reactome/release/downloaddirectory/GenerateGOAnnotationFile/CellularComponentAnnotationBuilder.java

```diff
@@ -2,88 +2,180 @@
+            ReactomeJavaConstants.Reaction,
+            Arrays.asList(
+                ReactomeJavaConstants.input, ReactomeJavaConstants.output, ReactomeJavaConstants.catalystActivity
+            )
+        );
+        classToClassAttributesForFollowingInstructions.put(
+            ReactomeJavaConstants.CatalystActivity,
+            Arrays.asList(ReactomeJavaConstants.physicalEntity)
+        );
+        classToClassAttributesForFollowingInstructions.put(
+            ReactomeJavaConstants.Complex,
+            Arrays.asList(ReactomeJavaConstants.hasComponent)
+        );
+        classToClassAttributesForFollowingInstructions.put(
+            ReactomeJavaConstants.EntitySet,
+            Arrays.asList(ReactomeJavaConstants.hasMember)
+        );
+        classToClassAttributesForFollowingInstructions.put(
+            ReactomeJavaConstants.Polymer,
+            Arrays.asList(ReactomeJavaConstants.repeatedUnit)
+        );
+
+        return classToClassAttributesForFollowingInstructions;
+    }
+
+    private static ClassAttributeFollowingInstruction getClassAttributeFollowingInstruction(
+        String className, List<String> classAttributes
+    ) {
+        final List<String> reverseClassAttributes = Collections.emptyList();
+
+        return new ClassAttributeFollowingInstruction(className, classAttributes, reverseClassAttributes);
+    }
+
+    private static String getAnyIssueForCellularComponentDisqualification(GKInstance protein) throws Exception {
+        String issueDisqualifyingProtein = getAnyIssueForAnnotationDisqualification(protein);
+        if (issueDisqualifyingProtein.isEmpty() && hasSpeciesWithAlternativeGOComponent(protein)) {
+            issueDisqualifyingProtein = protein.getExtendedDisplayName() + " is from a species with an alternative " +
+                "GO compartment, skipping GO annotation";
         }
+        return issueDisqualifyingProtein;
+    }
+
+    private static boolean hasSpeciesWithAlternativeGOComponent(GKInstance protein) throws Exception {
+        return speciesWithAlternateGOCompartment.contains(getTaxonIdentifier(protein));
+    }
+
+    /**
+     * Retrieves the Cellular Compartment accession associated with this protein and then calls the GOA line generator.
+     * @param protein -- GKInstance, Individual protein instance from the retrieved proteins.
+     * @param reactionlikeEvent -- GKInstance, Parent ReactionlikeEvent with which the protein is associated.
+     * @throws Exception -- MySQLAdaptor exception.
+     */
+    private static String generateGOCellularCompartmentLine(GKInstance protein, GKInstance reactionlikeEvent)
+        throws Exception {
+
+        String goCellularCompartmentAccession = getCellularCompartmentGOAccession(protein);
+        if (goCellularCompartmentAccession.isEmpty()) {
+            logger.info(protein.getExtendedDisplayName() + " has no Cellular Compartment accession, " +
+                "skipping GO annotation");
+            return "";
```

Comment:
Early returns in methods can be confusing.

Reply:
I prefer this method to deeply nested blocks and introducing temp variables for the return value.  This is something I usually do in my code to exit early with guard clauses when a condition for the core logic is not met.  It is an accepted refactoring that is often encouraged: https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

Reply:
Ok. Tastes differ.